script.js

```diff
@@ -1,12 +1,182 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
-}
+//variables:
+const rootElem = document.getElementById("root");
+const showRoot = document.getElementById("show-root");
+const lable = document.getElementById("lable");
+const allEpisodes = getAllEpisodes();
+const select = document.querySelector("#select");
+const searchBar = document.querySelector("#search-bar");
+const selectShow = document.querySelector("#select-show");
+const button = document.querySelector("#btn");
+let allShows = getAllShows();
+let episodeList;
+let allFetchEpisodes ='';
+
 
+//showing all episodes
+ function setup() {
+   makePageForShow(allShows);
+      
+ }
+
+//making page for each episode
 function makePageForEpisodes(episodeList) {
+  showRoot.innerText="";
+  let output = "";
+  episodeList.forEach((episode) => {
+    let seasonStr = episode.season.toString().padStart(2, "0");
+    let episodeStr = episode.number.toString().padStart(2, "0");
+    output += `
+    <div style="background-color:white; width:250px;border-radius:4px;">
+    <h1 style="color:black;">${episode.name} - S${seasonStr}E${episodeStr}</h1>
+    <img src="${episode.image.medium}"></img>
+    <div><p>${episode.summary}</p></div>
+    </div>
+    `;
+  });
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  rootElem.innerHTML = output;
+  let episodeNumber = document.createElement("div");
+  episodeNumber.textContent = `Got ${episodeList.length} episode(s)`;
+  rootElem.appendChild(episodeNumber);
+  return output;
+}
+
+window.onload = setup();
+
+// acsses to input's value
+const input = document.querySelector("#search-bar");
+input.addEventListener("keyup", function (e) {
+  let result = "";
+  const inputValue = e.target.value;
+  const allEpisodes = getAllEpisodes();
```

Comment:
You have done an amazing job fetching the episodes when you select one from the dropdown. The issue here is that when you do a search, you delete all the fetched episodes and get all the Game of Throne ones. 

To fix this, you will have to remove `const allEpisodes = getAllEpisodes();` and then on line 53 do a filter `allFetchEpisodes`, which you are creating within the `fetchEpisodesForShow` function 🎉 

Hopefully that makes sense, let me know if it does not and we can go through it together 🤝